estate/models/estate_property_offer.py

```diff
@@ -15,3 +15,30 @@ class EstatePropertyOffer(models.Model):
     )
     partner_id = fields.Many2one("res.partner", required=True)
     property_id = fields.Many2one("estate.property", required=True)
+    validity = fields.Integer(default=7, string="Validity (days)")
+    date_deadline = fields.Date(compute="_compute_date_deadline", inverse="_inverse_date_deadline", string="Deadline")
+
+    @api.depends("validity")
+    def _compute_date_deadline(self):
+        for record in self:
+            if not record.create_date:
+                create_date = fields.Date.today()
+            else:
+                create_date = record.create_date.date()
+            record.date_deadline = fields.Date.add(create_date, days=record.validity)
+
+    def _inverse_date_deadline(self):
+        for record in self:
+            record.validity = (record.date_deadline - record.create_date.date()).days
+
+    def set_status_accepted(self):
+        for record in self:
+            if sum(record.property_id.offer_ids.mapped(lambda x: x.status == "accepted")):
```

Comment:
Why you sum this? You could do:

```suggestion
            if "accepted" in record.property_id.offer_ids.mapped("status"):
```